Sources/epoch_code/compile/EPOCH_unitSpawn.sqf

```diff
@@ -12,34 +12,42 @@
     Github:
     https://github.com/EpochModTeam/Epoch/tree/master/Sources/epoch_code/compile/EPOCH_unitSpawn.sqf
 */
-private ["_unit","_sapperNum","_config","_bomb","_targetPos","_grp","_driver","_index","_nonJammer","_nonTrader","_jammers","_jammerRange","_restricted","_disableAI"];
-params ["_unitClass"];
-
+private ["_unit","_sapperNum","_config","_targetPos","_grp","_driver","_index","_nonJammer","_nonTrader","_jammers","_jammerRange","_restricted","_disableAI"];
+params ["_unitClass","_trgt","_doVariable"];
+private _bomb = objNull;
+if(isNil "_doVariable")then{_doVariable=false;};
+if(isNil "_trgt")then{_trgt=player;};
+diag_log format["Epoch: ADMIN: Antagonist %1 Spawning on %2. Do variable: %3.", _unitClass, _trgt, _doVariable];
 if(random 100 < 6)then{
 	[] execFSM "\x\addons\a3_epoch_code\System\Event_Air_Drop.fsm";
 };
+if(_doVariable)then{_unitClass call EPOCH_unitSpawnIncrease;};//Assumes Antagonist is required.
 
 _index = EPOCH_spawnIndex find _unitClass;
-if (count(player nearEntities[_unitClass, 800]) >= (EPOCH_playerSpawnArray select _index)) exitWith{};
+if (!_doVariable && (count(_trgt nearEntities[_unitClass, 800]) >= (EPOCH_playerSpawnArray select _index))) exitWith{};
 
 _nonJammer = ["CfgEpochClient", "nonJammerAI", ["B_Heli_Transport_01_F","PHANTOM","Epoch_Cloak_F"]] call EPOCH_fnc_returnConfigEntryV2;
 _nonTrader = ["CfgEpochClient", "nonTraderAI", ["B_Heli_Transport_01_F","PHANTOM","Epoch_Cloak_F","GreatWhite_F"]] call EPOCH_fnc_returnConfigEntryV2;
 _nonTraderAIRange = ["CfgEpochClient", "nonTraderAIRange", 150] call EPOCH_fnc_returnConfigEntryV2;
 
 _unit = objNull;
 
-_targetPos = getPosATL player;
+_targetPos = getPosATL _trgt;
 _targetPos set [2,0];
 
 _jammers = [];
 _config = 'CfgEpochClient' call EPOCH_returnConfig;
 _jammerRange = getNumber(_config >> "buildingJammerRange");
 _jammers = nearestObjects[_targetPos, ["PlotPole_EPOCH"], _jammerRange];
-if(count _jammers > 0 && (_unitClass in _nonJammer))exitWith{};
+if(count _jammers > 0 && !(_unitClass in _nonJammer))exitWith{
+if(_doVariable)then{["<t size='1.6' color='#99ffffff'>Not allowed near a base - Please respawn !</t>", 5] call Epoch_dynamicText;};
+};
 
 _restricted = [];
 _restricted = nearestObjects [_targetPos, ["ProtectionZone_Invisible_F"], _nonTraderAIRange];
-if(count _restricted > 0 && (_unitClass in _nonTrader))exitWith{};
+if(count _restricted > 0 && !(_unitClass in _nonTrader))exitWith{
```

Comment:
I don't believe I have changed that code. Is a bit misleading but essentially the array of unit classes are allowed to spawn by Jammers / Traders. Badly worded variables and double negatives, my bad :) We want to block all else.